msg_helpers.go

```diff
@@ -681,3 +681,135 @@ func packDataDomainNames(names []string, msg []byte, off int, compression compre
 	}
 	return off, nil
 }
+
+var aplAddrLenToFamily = map[int]uint16{
```

Comment:
Just use a switch statement for these. I'm pretty sure we don't use maps anywhere else.

Reply:
This is used in four different places so there will be some duplication but I can do that.